extensions/git/package.nls.json

```diff
@@ -143,6 +143,8 @@
 	"config.untrackedChanges.mixed": "All changes, tracked and untracked, appear together and behave equally.",
 	"config.untrackedChanges.separate": "Untracked changes appear separately in the Source Control view. They are also excluded from several actions.",
 	"config.untrackedChanges.hidden": "Untracked changes are hidden and excluded from several actions.",
+	"config.sshPrivateKeyPath": "Path of the SSH private key associated with your remote.",
```

Comment:
Is this needed even if the ssh config file includes proper Host entries?

Reply:
Also, what happens if the ssh-agent already has some keys added?

Reply:
Also see https://github.com/microsoft/vscode/issues/13680#issuecomment-285961890. It might be relevant.

Reply:
> Is this needed even if the ssh config file includes proper Host entries?

It can be avoided, but it would cost more substantial changes to the extension code, so I am waiting for reviews before doing it. I think the best way to use the SSH config file would not be to parse it (that would add complexity to the extension) but to read the output of the git commands and to handle any password prompt there. However, that would mean changing the `exec` function since it uses `child_process` which is unable to read such prompts, but replacing it with `node-pty` as I did here would not be enough since we would lose the stdout / stderr distinction.

> Also, what happens if the ssh-agent already has some keys added?

If the ssh-agent already has some keys added, nothing changes : you can add the same key twice without side effects.



> Also see [#13680 (comment)](https://github.com/microsoft/vscode/issues/13680#issuecomment-285961890). It might be relevant.

I agree that this could be an interesting feature. Git Bash's agents store their `SSH_AUTH_SOCK` in `%TEMP%\ssh-*` folders, so we could check them before starting the agent. I chose not to do it at first because I read that OpenSSH was now installed by default on Windows 10, and as you noted in your comment its agent is available to all processes. However I guess there is still a number of people who do not use it and would therefore be interested in this.

Reply:
Thanks for clarifying. Let's wait for some maintainers to get around to reviewing this.

I think it's better to get something in first and then iterate on feedback rather than holding things up to achieve perfection.